lib/AST/LayoutConstraint.cpp

```diff
@@ -58,44 +39,23 @@ LayoutConstraintInfo getLayoutConstraintInfo(StringRef ID) {
   return LayoutConstraintInfo(LayoutConstraintKind::UnknownLayout);
 }
 
-/// Checks if a given TypeRepr is a layout constraint type.
-bool isLayoutConstraintType(TypeRepr *TyR) {
-  return getLayoutConstraintInfo(TyR).isKnownLayout();
-}
-
-/// Checks if a given TypeRepr is a layout constraint and returns this
-/// constraint. If ID does not match any known layout constrains,
-/// returns UnknownLayout.
-LayoutConstraintInfo getLayoutConstraintInfo(TypeRepr *TyR) {
-  auto SimpleIdentType = dyn_cast<SimpleIdentTypeRepr>(TyR);
-  if (!SimpleIdentType)
-    return LayoutConstraintInfo(LayoutConstraintKind::UnknownLayout);
-
-  auto TypeName = SimpleIdentType->getIdentifier().str();
-  return getLayoutConstraintInfo(TypeName);
-}
-
-/// Checks if a given Type is a layout constraint and returns this
-/// constraint. If ID does not match any known layout constrains,
-/// returns UnknownLayout.
-LayoutConstraintInfo getLayoutConstraintInfo(Type Ty) {
-  auto CanTy = Ty->getCanonicalType();
-  if (auto Archetype = dyn_cast<ArchetypeType>(CanTy.getPointer())) {
-    auto Protocols = Archetype->getConformsTo();
-    for (auto Proto : Protocols) {
-      auto LayoutInfo = getLayoutConstraintInfo(Proto->getDeclaredType());
-      if (LayoutInfo.isKnownLayout())
-        return LayoutInfo;
-    }
-    return LayoutConstraintInfo(LayoutConstraintKind::UnknownLayout);
+StringRef LayoutConstraintInfo::getName() const {
+  switch (getKind()) {
+  case LayoutConstraintKind::UnknownLayout:
+    return "_UnknwonLayout";
```

Comment:
Typo: Unknwon

You might want to try using a .def file to define layout constraint kinds, like we do for types and decls etc.